### ISSUE TITLE
Inconsistent ordering of option aliases in help text

### ISSUE THREAD

**Issue by seanamosw** [dotnet/command-line-api#860]
Say we have some options:

```csharp
new Command("command")
{
    new Option<string>(new [] { "--option1", "-x" }),
    new Option<string>(new [] { "--option2", "-a" }),
    new Option<string>(new [] { "--option3", "-b" })
}
```

The options help text output for this is:

```
Options:
  --option1, -x <option1>                             
  -a, --option2 <option2>                             
  -b, --option3 <option3>  
```

I would have expected the short prefix aliases to always come first, with the full alias to come second. Instead the aliases are just alphabetically ordered.

I think the issue is in the code in https://github.com/dotnet/command-line-api/blob/master/src/System.CommandLine/Help/HelpBuilder.cs :
```csharp
            var rawAliases = symbol
                             .RawAliases
                             .Select(r => r.SplitPrefix())
                             .OrderBy(r => r.alias)
                             .ThenBy(r => r.prefix)
                             .GroupBy(t => t.alias)
                             .Select(t => t.First())
                             .Select(t => $"{t.prefix}{t.alias}");
```

Swapping the `OrderBy` and `ThenBy`, order by prefix then alias, seems to make more sense to me.

**Comment by Keboo** [comment]
I would agree with sorting by prefix before alias

**Comment by jonsequitur** [comment]
The order in which the options appear in help is the order in which they're added to the command. This is the simplest way to allow people to provide a custom order, as many command line tools do. I'll add a test to clarify this. Here's the upshot:

```csharp
var command = new RootCommand
{
    new Option(new[] { "--first", "-f" }),
    new Option(new[] { "--second", "-s" }),
    new Option(new[] { "--third" }),
    new Option(new[] { "--last", "-l" })
};

_helpBuilder.Write(command);
var help = _console
           .Out
           .ToString()
           .Split(new[] { '\n', '\r' }, StringSplitOptions.RemoveEmptyEntries)
           .Select(l => l.Trim());

help.Should().ContainInOrder(
    "-f, --first",
    "-s, --second",
    "--third",
    "-l, --last");
```

The code you pointed to is sorting the aliases within a given option's help output, so that the short aliases appear before the long aliases for each option.

**Comment by seanamosw** [comment]
I agree, the order in which options are added to a command are their sort order for help text, this seems correct and behaved the way I expected.

I am referring to the ordering of aliases for a given option. You can see in the example I posted above, option aliases just seem to be sorted alphabetically which I didn't expect. This results in option help which can look like:

```
Options:
  --option1, -x <option1>
  -a, --option2 <option2>
  --option3, -p <option3>
  -b, --option4 <option2>
```

I expected them to either be in the order they were added or if they were sorted, to have consistent sorting of the prefixes.

If the short alias is the same as the first letter of the long alias, you don't encounter this issue.

**Comment by jonsequitur** [comment]
Sorry, I misunderstood the issue initially. 

The fix is here: #871
